source/dhtslib/coordinates.d

```diff
@@ -256,6 +271,99 @@ struct Coordinates(CoordSystem cs)
         }
     }
 
+    /// Convert coordinate to another based system using shortcuts
+    auto to(T: ZBHO)()
+    {
+        return this.to!(CoordSystem.zbho);
+    }
+
+    /// Convert coordinate to another based system using shortcuts
+    auto to(T: OBHO)()
+    {
+        return this.to!(CoordSystem.obho);
+    }
+
+    /// Convert coordinate to another based system using shortcuts
+    auto to(T: ZBC)()
+    {
+        return this.to!(CoordSystem.zbc);
+    }
+    
+    /// Convert coordinate to another based system using shortcuts
+    auto to(T: OBC)()
+    {
+        return this.to!(CoordSystem.obc);
+    }
+
+    /// make a new coordinate pair with a value of 
+    /// this.start + off and this.end + off
+    auto offset(T)(T off)
+    if(isIntegral!T)
+    {
+        return Coordinates!(cs)(cast(long)(this.start + off), cast(long)(this.end + off));
+    }
+
+    /// intersection of two regions
+    auto intersectImpl(Coordinates!cs other)
+    {
+        if(!this.isOverlap(other)){
+            return Coordinates!(cs).init;
+        }
+        return Coordinates!(cs)(
+            this.getMaxStart(other),
+            this.getMinEnd(other)
+            );
+    }
+
+    /// union of two regions
+    auto unionImpl(Coordinates!cs other)
+    {
+        if(!this.isOverlap(other)){
+            return Coordinates!(cs).init;
```

Comment:
So yes a true union would be an array of both intervals. I figured that wouldn't be preferred. I wasn't sure what is normal for set logic, throw error or return an empty interval.

Reply:
Given these constraints, it may not be reasonable to implement union on ranges. An alternative interpretation of union could be the range `[min(A,B), max(A,B)]` -- what do youy think?

I definitely agree with intersection operation though

Reply:
There's actually no reason we can't return an array of results, now that I think of it. However this invokes the GC if we use a dynamic array. How free of the GC  are we in the rest of the codebase?

Reply:
Should be pretty GC free. No usage of arrays, all types can live on the stack with the exception of the `string` in `ChromCoordinates`.

Reply:
> Given these constraints, it may not be reasonable to implement union on ranges. An alternative interpretation of union could be the range `[min(A,B), max(A,B)]` -- what do youy think?

?